search/processor.cpp

```diff
@@ -717,27 +717,39 @@ void Processor::SearchPlusCode()
   string query(m_query);
   strings::Trim(query);
 
-  string code;
-
   if (openlocationcode::IsFull(query))
   {
-    code = query;
+    openlocationcode::CodeArea const area = openlocationcode::Decode(query);
+    m_emitter.AddResultNoChecks(
+            m_ranker.MakeResult(RankerResult(area.GetCenter().latitude, area.GetCenter().longitude),
+                                true /* needAddress */, false /* needHighlighting */));
   }
   else if (openlocationcode::IsShort(query))
   {
-    if (!m_position)
-      return;
-    ms::LatLon const latLon = mercator::ToLatLon(*m_position);
-    code = openlocationcode::RecoverNearest(query, {latLon.m_lat, latLon.m_lon});
-  }
 
-  if (code.empty())
-    return;
+    ms::LatLon const latLonFromView = mercator::ToLatLon(m_viewport.Center());
+    ms::LatLon const latLonFromPos = mercator::ToLatLon(*m_position);
+
+    string codeFromView = openlocationcode::RecoverNearest(query, {latLonFromView.m_lat, latLonFromView.m_lon});
+    string codeFromPos = openlocationcode::RecoverNearest(query, {latLonFromPos.m_lat, latLonFromPos.m_lon});
+
+    if (codeFromView != codeFromPos)
+    {
+      openlocationcode::CodeArea const areaFromView = openlocationcode::Decode(codeFromView);
+      m_emitter.AddResultNoChecks(
+              m_ranker.MakeResult(RankerResult(areaFromView.GetCenter().latitude, areaFromView.GetCenter().longitude),
+                                  true /* needAddress */, true /* needHighlighting */));
+    }
+
+    if (!latLonFromPos.EqualDxDy({0, 0}, 0.00001)) // TODO: better way to check if no location found?
```

Comment:
for reviewers: when we have no location fix, m_position is `(0.0, 0.0)`. I'm not sure how else to check if no location has been found yet.

Reply:
Without a location fix, m_position optional should not be set (should be false). Can you please debug why/when is it set to zero?

Reply:
i'm not sure how to debug the CPP code unfortunately, logs don't seem to be written to the android log feed. i think this might have been caused by latLonFromPos initialising as 0,0 when m_location isn't available though, not an issue with m_location.